src-ocaml/baijiu_blake2b.ml

```diff
@@ -335,13 +338,19 @@ module Unsafe : S = struct
     with_outlen_and_key ~blit:By.blit_from_bigstring outlen key off len
 
   let unsafe_get ctx =
-    let res = By.make 64 '\x00' in
+    let res = By.make default_param.digest_length '\x00' in
     increment_counter ctx (Int64.of_int ctx.buflen) ;
     set_lastblock ctx ;
     By.fill ctx.buf ctx.buflen (128 - ctx.buflen) '\x00' ;
     compress ~le64_to_cpu:By.le64_to_cpu ctx ctx.buf 0 ;
     for i = 0 to 7 do
       By.cpu_to_le64 res (i * 8) ctx.h.(i)
     done ;
-    By.sub res 0 ctx.outlen
+    if ctx.outlen < default_param.digest_length
+    then By.sub res 0 ctx.outlen
+    (* XXX(dinosaure): should never appear! *)
+    else if ctx.outlen > default_param.digest_length
+    then ( let res' = By.make ctx.outlen '\x00' in
+           By.blit res 0 res' 0 default_param.digest_length ; res' )
```

Comment:
In this case (`ctx.outlen > default_param.digest_length`) we will be returning a `bytes` variable with `ctx.outlen - default_param_digest_length` x `\x00` right-padded.
It looks a bit weird to me, but I am not familiar with the intended semantics - is that the intention?

If it's intentional, would it make sense to have a unit test for this case?

Reply:
This case is protected by something else which is when you try to make a new `BLAKE2{S,B}` implementation with a smaller `outlen`.

If you see: https://github.com/mirage/digestif/pull/85/files#diff-0896e018d0139e68e8fa2d2bff425989R324

It's not possible to make a `BLAKE2{S,B}  where `outlen` will be upper than `default_param.digest_length`. This is intentional but the case can not be tested where `Make_BLAKE2{S,B}(struct let digest_size = 128 end)` will fail your program instantaneously. So, yeah it's really wierd but it's why I noticed that this case should never appear.

Reply:
wouldn't it suffice to have an assertion / a `failwith` in that case?